### DISCUSSION TITLE
Empty App Service list when adding new App Service certificate

### DISCUSSION THREAD

**Discussion by juhovainio** [rcl-ssl/docs#21] [Q&A] [answered]
I faced a rather odd bug while trying to add an auto renew cert to a pre-existing App Service under our Azure Subscription. The lists of App Service Plans and Site Names appear to be empty, even after I've selected the correct Subscription and even though I have added RCL application all the necessary rights as outlined in the RCL Docs.

![image](https://user-images.githubusercontent.com/3143199/137038191-9370b7d1-918a-4b03-9761-f1b7d997e944.png)

I'm also able to produce an error while trying to create a new Key Vault certificate so I was wondering if this might still have possibly have something to do with some access right configurations? The latter goes with the following error code thrown: `|afa39853-451daadf9c48e43f`

Thanks in advance :)

**Comment by rcladmin** [comment, marked as answer]
Did you try logging out of the account , clearing the cache in the browser and logging back in? Is the user a 'owner' or 'contributor' to the subscription? Did you properly set access control for the user (https://docs.rclapp.com/authorization/access-control-user.html#set-access-control-for-the-aad-user)?

**Reply by juhovainio** [reply]
Thanks for the reply, I tried the refresh method but eventually I was able to get this working by creating a fresh user account under the same AAD domain as the Subscription and redoing the steps I had done with my previous one. I guess the account must have been a Guest user, I can't figure out any other reason.

Well now that's working my next problem is with getting the API Client ID working in order to perform the Automatic Renewals but it's now throwing this error which is not documented anywhere. I followed the AutoRenew Function documentation until this stage and double checked that I had done all the listed steps. I tried checking the developer console but could not see any other errors after adding and submitting the AAD Application Client ID.

![image](https://user-images.githubusercontent.com/3143199/138360748-5e0c99d1-2d43-4771-aba6-b66ace79027c.png)

**Reply by rcladmin** [reply]
That's strange. Does the app allow you to enter the Client ID or does it throw this error? Are you logged in as a co-admin?

**Reply by rcladmin** [reply]
Please try to enter your Client ID now and see if it works.

**Reply by juhovainio** [reply]
Yes, the error was thrown after the Client ID was entered. But testing this now things seem to be working correctly. Thanks! :)
